packages/plugins/plugin-table/src/TablePlugin.tsx

```diff
@@ -91,25 +97,20 @@ export const TablePlugin = (): PluginDefinition<TablePluginProvides> => {
         },
       },
       surface: {
-        component: ({ data, role }) => {
-          switch (role) {
-            case 'slide':
-              return isTable(data.slide) ? <TableContainer table={data.slide} /> : null;
-            case 'section':
-            case 'article':
-              return isTable(data.object) ? <TableContainer role={role} table={data.object} /> : null;
-            case 'complementary--settings': {
-              if (data.subject instanceof TableType) {
-                const table = data.subject;
-                return { node: <TableViewEditor table={table} /> };
-              }
-
-              return null;
-            }
-            default:
-              return null;
-          }
-        },
+        definitions: () => [
+          createSurface({
+            id: `${TABLE_PLUGIN}/table`,
+            role: ['article', 'section', 'slide'],
+            filter: (data): data is { subject: TableType } => data.subject instanceof TableType,
```

Comment:
i wonder if target would be a better label than subject?

Reply:
can you explain target? subject is the thing that's made the most sense to me as a generic term for an arbitrary thing passed in for the surface to render: it's the subject of the surface